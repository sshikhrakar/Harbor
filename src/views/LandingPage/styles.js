import { StyleSheet } from 'react-native';

import fonts from '../../config/fonts';
import colors from '../../config/colors';

const styles = StyleSheet.create({
  landingPageContainer: {
    flex: 1,
    paddingVertical: 16,
    backgroundColor: colors.SILVER,
  },

  logoContainer: {
    flex: 1,
    paddingVertical: 28,
    paddingHorizontal: 18,
    alignItems: 'center',
    justifyContent: 'center',
  },

  landingPageLogo: {
    flex: 1,
    resizeMode: 'contain',
  },

  greetingContainer: {
    justifyContent: 'flex-start',
    alignItems: 'center',
  },

  mainText: {
    fontSize: 36,
    alignSelf: 'center',
    fontFamily: fonts.primary.REGULAR,
  },

  subText: {
    fontSize: 18,
    fontFamily: fonts.primary.LIGHT,
  },

  loginContainer: {
    flex: 3,
    padding: 24,
    justifyContent: 'flex-start',
  },

  loginText: {
    fontSize: 18,
    alignSelf: 'center',
    fontFamily: fonts.primary.REGULAR,
  },

  link: {
    fontSize: 14,
    alignSelf: 'center',
    fontFamily: fonts.primary.REGULAR,
    color: colors.SOFT_BLUE,
  },

  disabledLink: {
    fontSize: 14,
    alignSelf: 'center',
    fontFamily: fonts.primary.REGULAR,
    color: colors.ASBESTOS,
  },

  error: {
    fontSize: 14,
    alignSelf: 'center',
    textAlign: 'center',
    color: colors.ERROR,
    fontFamily: fonts.primary.LIGHT,
  },

  emailPasswordContainer: {
    flex: 2,
    justifyContent: 'space-around',
  },

  textInputField: {
    height: 44,
    borderWidth: 0.8,
    borderRadius: 6,
    paddingHorizontal: 10,
    borderColor: colors.ASBESTOS,
    fontFamily: fonts.primary.REGULAR,
  },

  loginButtonContainer: {
    flex: 1,
    paddingVertical: 10,
  },

  button: {
    flex: 0.8,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: colors.ASBESTOS,
  },

  loginButtonText: {
    fontSize: 20,
    marginVertical: 5,
    color: colors.SILVER,
    fontFamily: fonts.primary.LIGHT,
  },

});

export default styles;
