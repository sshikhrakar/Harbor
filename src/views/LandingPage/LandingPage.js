import React from 'react';
import {
  Text,
  View,
  Image,
  Modal,
  TextInput,
} from 'react-native';
import PropTypes from 'prop-types';

import { TextLink, ButtonWithSpinner } from '../../components';

import styles from './styles';
import images from '../../config/images';
import colors from '../../config/colors';

function LandingPage(props) {

  const {
    emailText,
    errorText,
    isLoggingIn,
    passwordText,
    setEmailText,
    setPasswordText,
    onLoginButtonPress,
    onCreateAccountPress,
    onCreateAccountModalClose,
    createAccountModalVisibility,
  } = props;

  return (
    <View style={ styles.landingPageContainer }>

      <View style={ styles.logoContainer}>
        <Image
          style={ styles.landingPageLogo }
          source={ images.harborLogo }
          tintColor={ colors.GREY }
        />
      </View>

      <View style={ styles.greetingContainer}>
        <Text style={ styles.mainText }>
          HARBOR
        </Text>
        <Text style={ styles.subText }>
          Share Android builds of React Native apps
        </Text>
      </View>

      <Modal
        animationType={ 'slide' }
        transparent={ false }
        visible={ createAccountModalVisibility }
        onRequestClose={ onCreateAccountModalClose }
      >
        <View>
          <Text>Hello world</Text>
        </View>
      </Modal>


      <View style={ styles.loginContainer }>
        <Text style={ styles.loginText }>
          LOGIN TO CONTINUE
        </Text>

        {
          errorText ?
            <Text style={ styles.error }>
              { errorText }
            </Text>
            : null
        }

        <View style={ styles.emailPasswordContainer }>
          <TextInput
            value={ emailText }
            placeholder={ 'E-mail' }
            onChangeText={ setEmailText }
            style={ styles.textInputField }
            keyboardType="email-address"
            underlineColorAndroid={ 'transparent' }
          />
          <TextInput
            value={ passwordText }
            secureTextEntry= { true }
            placeholder={ 'Password' }
            style={ styles.textInputField }
            onChangeText={ setPasswordText }
            underlineColorAndroid={ 'transparent' }
          />
        </View>

        <ButtonWithSpinner
          isLoading={ isLoggingIn }
          text="LOGIN"
          onPress={ onLoginButtonPress }
        />

        <TextLink
          onPress={ onCreateAccountPress }
          text="Or, Create an Account."
          style={ styles.link }
        />

      </View>
    </View>
  );

}

LandingPage.propTypes = {
  isLoggingIn: PropTypes.bool.isRequired,
  emailText: PropTypes.string.isRequired,
  errorText: PropTypes.string.isRequired,
  setEmailText: PropTypes.func.isRequired,
  passwordText: PropTypes.string.isRequired,
  setPasswordText: PropTypes.func.isRequired,
  onLoginButtonPress: PropTypes.func.isRequired,
  onCreateAccountPress: PropTypes.func.isRequired,
  onCreateAccountModalClose: PropTypes.func.isRequired,
  createAccountModalVisibility: PropTypes.bool.isRequired,
};


export default LandingPage;
