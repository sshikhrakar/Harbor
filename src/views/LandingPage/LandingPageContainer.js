import {
  branch,
  compose,
  withState,
  lifecycle,
  withHandlers,
  renderComponent,
} from 'recompose';
import { connect } from 'react-redux';

import HomePage from '../HomePage';
import LandingPage from './LandingPage';

import { validators } from '../../utils';
import { loginViaEmail } from '../../actions/authActions';

const mapStateToProps = state => ({
  isLoggedIn: state.auth.isLoggedIn,
  isLoggingIn: state.ui.auth.isLoggingIn,
  hasLoginErrored: state.ui.auth.hasLoginErrored,
});

const mapDispatchToProps = {
  loginViaEmail,
};

const enhance = compose(
  connect(mapStateToProps, mapDispatchToProps),

  withState('errorText', 'setErrorText', ''),
  withState('passwordText', 'setPasswordText', '123456'),
  withState('emailText', 'setEmailText', 'srishanbhattarai@gmail.com'),
  withState('createNewAccountEmailText', 'setCreateNewAccountPasswordText', '123456'),
  withState('createAccountModalVisibility', 'setCreateAccountModalVisibility', false),
  withState('createNewAccountEmailText', 'setCreateNewAccountEmailText', 'srishanbhattarai@gmail.com'),

  branch(
    props => props.isLoggedIn,
    renderComponent(HomePage),
  ),

  withHandlers({
    onLoginButtonPress: props => () => {
      if (!validators.isValidEmail(props.emailText)) {
        props.setErrorText('Please enter a valid email address.');
      } else if (!validators.isValidPassword(props.passwordText)) {
        props.setErrorText('Please enter a valid password.');
      } else {
        props.setErrorText('');
        props.loginViaEmail(props.emailText, props.passwordText);
      }
    },

    onCreateAccountPress: props => () => {
      props.setCreateAccountModalVisibility(true);
    },

    onCreateAccountModalClose: props => () => {
      props.setCreateNewAccountEmailText('');
      props.setCreateNewAccountPasswordText('');
    },
  }),

  lifecycle({
    componentWillReceiveProps(nextProps) {
      if (nextProps.hasLoginErrored) {
        this.setState({
          errorText: 'Cannot login. Please make sure your credentials are valid.',
        });
      }
    },
  }),

);

export default enhance(LandingPage);
