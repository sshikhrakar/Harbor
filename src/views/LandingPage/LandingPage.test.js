import React from 'react';
import renderer from 'react-test-renderer';

import LandingPage from './LandingPage';

describe('VIEWS: LandingPage', () => {

  it('should render correctly', () => {
    const tree = renderer.create(
      <LandingPage
        isLoggingIn={ false }
        emailText="foobar@gmail.com"
        errorText=""
        setEmailText={ (text) => this.setState({ emailText: text }) }
        passwordText="123456"
        setPasswordText={ (text) => this.setState({ passwordText: text }) }
        onLoginButtonPress={ () => true }
        onCreateAccountPress={ () => true }
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
