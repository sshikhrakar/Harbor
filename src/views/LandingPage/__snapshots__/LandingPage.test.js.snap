// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VIEWS: LandingPage should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ecf0f1",
      "flex": 1,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 18,
        "paddingVertical": 28,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "flex": 1,
          "resizeMode": "contain",
        }
      }
      tintColor="#95a5a6"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "fontFamily": "Lato-Regular",
          "fontSize": 36,
        }
      }
    >
      HARBOR
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Lato-Light",
          "fontSize": 18,
        }
      }
    >
      Share Android builds of React Native apps
    </Text>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={true}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        Hello world
      </Text>
    </View>
  </Modal>
  <View
    style={
      Object {
        "flex": 3,
        "justifyContent": "flex-start",
        "padding": 24,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "fontFamily": "Lato-Regular",
          "fontSize": 18,
        }
      }
    >
      LOGIN TO CONTINUE
    </Text>
    <View
      style={
        Object {
          "flex": 2,
          "justifyContent": "space-around",
        }
      }
    >
      <TextInput
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="E-mail"
        style={
          Object {
            "borderColor": "#7f8c8d",
            "borderRadius": 6,
            "borderWidth": 0.8,
            "fontFamily": "Lato-Regular",
            "height": 44,
            "paddingHorizontal": 10,
          }
        }
        underlineColorAndroid="transparent"
        value="foobar@gmail.com"
      />
      <TextInput
        onChangeText={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "#7f8c8d",
            "borderRadius": 6,
            "borderWidth": 0.8,
            "fontFamily": "Lato-Regular",
            "height": 44,
            "paddingHorizontal": 10,
          }
        }
        underlineColorAndroid="transparent"
        value="123456"
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingVertical": 10,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#7f8c8d",
            "flex": 0.8,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#ecf0f1",
              "fontFamily": "Lato-Light",
              "fontSize": 20,
              "marginVertical": 5,
            }
          }
        >
          LOGIN
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#3498db",
            },
            Object {
              "alignSelf": "center",
              "color": "#3498db",
              "fontFamily": "Lato-Regular",
              "fontSize": 14,
            },
          ]
        }
      >
         
        Or, Create an Account.
         
      </Text>
    </View>
  </View>
</View>
`;
